### PR TITLE
[SPARK-18410][STREAMING] Add structured kafka example

### PR DESCRIPTION
**Pull request by uncleGen**
## What changes were proposed in this pull request?

This PR provides structured kafka wordcount examples

### CODE REVIEW

**Comment by rxin** [comment]
@koeninger want to review this?

File: examples/src/main/java/org/apache/spark/examples/sql/streaming/JavaStructuredKafkaWordCount.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ *   <subscribe-type> There are three kinds of type, i.e. 'assign', 'subscribe',
+ *   'subscribePattern'.
+ *   |- <assign> Specific TopicPartitions to consume. Json string
+ *   |  {"topicA":[0,1],"topicB":[2,4]}.
+ *   |- <subscribe> The topic list to subscribe. A comma-separated list of
+ *   |  topics.
+ *   |- <subscribePattern> The pattern used to subscribe to topic(s).
+ *   |  Only one of "assign, "subscribe" or "subscribePattern" options can be
+ *   |  specified for Kafka source.
+ *   <topics> A list of one or more kafka topics to consume from. Please refer
```
Comment (koeninger) [inline examples/src/main/java/org/apache/spark/examples/sql/streaming/JavaStructuredKafkaWordCount.java]: This isn't really a list of topics for assign or subscribePattern. I'd say use a more general description, or stick to just subscribe

**Review by koeninger — COMMENTED** [review]
Generally looks OK to me.
One question - am I misremembering that we previously had issues building a single example jar that had both kafka 0.8 and 0.10 dependencies in it? What changed?

File: examples/src/main/java/org/apache/spark/examples/sql/streaming/JavaStructuredKafkaWordCount.java (hunk 1 above)
Comment (uncleGen) [inline examples/src/main/java/org/apache/spark/examples/sql/streaming/JavaStructuredKafkaWordCount.java]: get it

**Comment by koeninger** [comment]
LGTM

**Comment by uncleGen** [comment]
just rebase

**Comment by srowen** [comment]
Merged to master/2.1

**Pull request closed without merging**